studies/forms.py

```diff
@@ -441,7 +441,12 @@ class Meta:
 
     def clean_structure(self):
         try:
-            structure = json.loads(self.cleaned_data["structure"])
+            # Validate structure if not using generator
+            if not self.cleaned_data["use_generator"]:
+                structure = json.loads(self.cleaned_data["structure"])
+            else:
+                structure = {}
+
```

Comment:
Does this mean that any JSON in the protocol box will be overwritten if the researcher selects 'use generator' and saves? Probably not a big deal if so, but just checking because I think the way it works now is that any JSON in the protocol box is still saved when the researcher chooses to use the protocol generator. 
Would it work to just remove the `else` here or put the next few lines inside the `if`, or would that break things?

Reply:
The code the user enters is retained in the "exact_text" key of that dictionary.  This means there is no JSON for EFP to use, but if the user selects "use_generator" and saves, then the JSON is translated to dictionary, validated, and is available for EFP.  

We need the empty dictionary on 448 to hold the "exact_text" on 450.

Reply:
Ok thanks!